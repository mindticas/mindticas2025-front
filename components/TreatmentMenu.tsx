'use client';

import {
    SelectContent,
    SelectItem,
    SelectLabel,
    SelectTrigger,
    SelectValueText,
} from '@/components/ui/select';

import React, { useEffect, useState } from 'react';
import { useBookingContext } from '@/context/BookingContext';
import { getTreatments } from '@/services/TreatmentService';
import { Treatment } from '@/interfaces/treatment/Treatment';
import { SelectRoot, Text, createListCollection } from '@chakra-ui/react';

export default function TreatmentMenu() {
    const { treatment, setTreatment, treatmentDuration, setTreatmentDuration } =
        useBookingContext();
    const [treatments, setTreatments] = useState<Treatment[]>([]);
    const [error, setError] = useState<string | null>(null);

    useEffect(() => {
        const fetchTreatments = async () => {
            try {
                const data = await getTreatments();
                setTreatments(data);
                const durationData = treatments.find((t) => t.id === treatment);
                console.log(durationData);
            } catch (error) {
                setError(
                    'No se pudieron cargar los tratamientos. Inténtalo de nuevo más tarde.',
                );
            }
        };
        fetchTreatments();
    }, []);

    // get treatment duration when treatment changes
    useEffect(() => {
        getTreatmentDuration();
        console.log(treatment);
    }, [treatment]);

    // get treatment duration
    const getTreatmentDuration = () => {
        const durationData = treatments.find((t) => t.id === treatment);
        console.log(durationData);
        if (durationData) {
            setTreatmentDuration(durationData.duration);
        } else {
            setTreatmentDuration(0);
        }
    };

    const handleTreatmentSelect = (selectedValue: string) => {
        const treatmentId = parseInt(selectedValue);
        setTreatment(treatmentId);
    };

    const getSelectedTreatmentName = () => {
        const selectedTreatment = treatments.find((t) => t.id === treatment);
        return selectedTreatment?.name ?? 'Servicio a realizar';
    };

    return (
        <>
            <div>{error && <p style={{ color: 'red' }}>{error}</p>}</div>
            <SelectRoot
                onValueChange={(e) => handleTreatmentSelect(e.value.toString())}
                size='lg'
                maxW='sm'
                p={3}
                collection={createListCollection({ items: treatments })}
            >
                <SelectLabel fontWeight='bold' ps='1' fontSize='lg'>
                    Servicio a realizar
                </SelectLabel>
                <SelectTrigger border='black solid 1px'>
                    {getSelectedTreatmentName()}
                    <SelectValueText />
                </SelectTrigger>
                <SelectContent backgroundColor='white'>
                    {treatments.map(({ id, price, name }) => (
                        <SelectItem
                            cursor='pointer'
                            _hover={{ backgroundColor: 'gray.100' }}
                            backgroundColor='white'
                            item={id.toString()}
                            key={id}
                            p={2}
                            data-state={
                                treatment === id ? 'checked' : 'unchecked'
                            }
                            // disable the treatment if the booking is already made
                        >
                            {name}
                            <Text fontWeight='bold'>{`$${price} `}</Text>
                        </SelectItem>
                    ))}
                </SelectContent>
            </SelectRoot>
        </>
    );
}
