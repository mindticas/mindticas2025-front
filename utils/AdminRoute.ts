import Layout from '@/components/Layout';
import { Box } from '@chakra-ui/react';

export class AdminRoute {
    // Fixed route for admin
    private static route: string = '/admin';

    // Returns the Layout component if the pathname contains '/admin', otherwise returns box
    static getLayout(pathname: string) {
        return pathname.includes(this.route) ? Layout : Box;
    }
}
