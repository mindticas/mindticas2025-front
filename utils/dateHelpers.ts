// Gets the name of the day according to the full calendar format
export const getDayName = (dateStr: string): string => {
    const [year, month, day] = dateStr.split('-').map(Number)
    const date = new Date(year, month -1, day)
    return date.toLocaleDateString('es-ES', { weekday: 'long'})
}
// Get the milliseconds in a day
export function addDays(date: Date, days: number): Date {
    const milisecondsPerDay = 24 * 60 * 60 * 1000; 
    return new Date(date.getTime() + days * milisecondsPerDay)
}