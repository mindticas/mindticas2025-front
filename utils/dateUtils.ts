import { DateTime } from "luxon";

interface DateTimeParts {
    date: string;
    time: string;
}

// * @returns Object with (YYYY-MM-DD) y time (HH:MM)

export const splitDateTimeFromISO = (
    isoString: string | null | undefined,
): DateTimeParts => {
    if (!isoString) return { date: '', time: '' };

    const dateObj = new Date(isoString);

    if (isNaN(dateObj.getTime())) {
        return { date: '', time: '' };
    }

    // format date (YYYY-MM-DD)
    const date = DateTime.fromJSDate(dateObj)
       .setZone('local')
        .toFormat('yyyy-MM-dd');

    // format time (HH:MM)
    const time = dateObj
        .toLocaleTimeString('es-ES', {
            hour: '2-digit',
            minute: '2-digit',
            hour12: false,
        })
        .replace(/\./g, '');

    return { date, time };
};
