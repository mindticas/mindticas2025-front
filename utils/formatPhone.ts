export default function formatPhone(phone: string): string {
    // Remove everything that is not digits
    const digits = phone.replace(/\D/g, '');
    // Applies format only if it has 10 digits
    if (digits.length === 10) {
        return `${digits.slice(0, 3)}-${digits.slice(3, 6)}-${digits.slice(6)}`;
    }
    return phone; // Returns the original if it doesn't match
}
