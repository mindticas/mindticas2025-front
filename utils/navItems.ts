import { BarChart, Calendar, Scissors, UserCog, Users } from 'lucide-react';

export const navItems = [
    { name: 'Citas', href: '/admin/', icon: Calendar },
    { name: 'Clientes', href: '/admin/clients', icon: Users },
    { name: 'Empleados', href: '/admin/employees', icon: UserCog },
    { name: 'Tratamientos', href: '/admin/treatments', icon: Scissors },
    { name: 'Reportes', href: '/admin/reports', icon: BarChart },
];
