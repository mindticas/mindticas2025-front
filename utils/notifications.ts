import { toaster } from '@/components/ui/toaster';

type ToastType = 'success' | 'error' | 'info' | 'warning';

interface ToastOption {
    duration?: number;
    title: string;
    description?: string;
}

const showToast = (type: ToastType, options: ToastOption) => {
    toaster.create({
        type,
        duration: options.duration || 5000,
        title: options.title,
        description: options.description,
    });
};

const TreatmentNotification = {
    createSuccess: () =>
        showToast('success', { title: 'Tratamiento creado con éxito' }),
    updateSuccess: () =>
        showToast('success', { title: 'Tratamiento actualizado con éxito' }),
    deleteSuccess: () =>
        showToast('success', { title: 'Tratamiento eliminado con éxito' }),
    createError: (message?: string) =>
        showToast('error', {
            title: message || 'Error al crear el tratamiento',
        }),
    updateError: (message?: string) =>
        showToast('error', {
            title: message || 'Error al actualizar el tratamiento0',
        }),
    deleteError: () =>
        showToast('error', { title: 'Error al eliminar el tratamiento' }),
    noChanges: () =>
        showToast('info', {
            title: 'No se detectaron cambios para actualizar',
        }),
};

export default TreatmentNotification;
