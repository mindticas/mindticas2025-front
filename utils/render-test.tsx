import { Provider } from '@/components/ui/provider';
import { BookingProvider } from '@/context/BookingContext';
import { render as rtlRender } from '@testing-library/react';

export function render(ui: React.ReactNode) {
    return rtlRender(<>{ui}</>, {
        wrapper: (props: React.PropsWithChildren) => (
            <Provider>
                <BookingProvider>{props.children}</BookingProvider>
            </Provider>
        ),
    });
}
