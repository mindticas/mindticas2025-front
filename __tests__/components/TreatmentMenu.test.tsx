import TreatmentMenu from '@/components/TreatmentMenu';
import { render } from '../../utils/render-test';

describe('HomePage Snapshot', () => {
    it('should match the snapshot', () => {
        const { asFragment } = render(<TreatmentMenu />);
        expect(asFragment()).toMatchSnapshot();
    });
});
