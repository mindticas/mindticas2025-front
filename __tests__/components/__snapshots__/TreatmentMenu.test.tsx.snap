// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomePage Snapshot should match the snapshot 1`] = `
<DocumentFragment>
  <script
    nonce=""
  >
    ((e,s,n,m,o,l,u,h)=&gt;{let a=document.documentElement,v=["light","dark"];function p(c){(Array.isArray(e)?e:[e]).forEach(y=&gt;{let k=y==="class",S=k&&l?o.map(f=&gt;l[f]||f):o;k?(a.classList.remove(...S),a.classList.add(c)):a.setAttribute(y,c)}),C(c)}function C(c){h&&v.includes(c)&&(a.style.colorScheme=c)}function d(){return window.matchMedia("(prefers-color-scheme: dark)").matches?"dark":"light"}if(m)p(m);else try{let c=localStorage.getItem(s)||n,y=u&&c==="system"?d():c;p(y)}catch(c){}})("class","theme","system",null,["light","dark"],null,true,true)
  </script>
  <div />
  <div
    class="chakra-select__root css-19aax7y"
    data-part="root"
    data-scope="select"
    dir="ltr"
    id="select::r0:"
  >
    <label
      class="chakra-select__label css-t80ohu"
      data-part="label"
      data-scope="select"
      dir="ltr"
      for="select::r0::select"
      id="select::r0::label"
    >
      Servicio a realizar
    </label>
    <div
      class="chakra-select__control css-17j1053"
      data-part="control"
      data-scope="select"
      data-state="closed"
      dir="ltr"
      id="select::r0::control"
    >
      <button
        aria-controls="select::r0::content"
        aria-expanded="false"
        aria-haspopup="listbox"
        aria-labelledby="select::r0::label"
        class="chakra-select__trigger css-1qa70k0"
        data-part="trigger"
        data-placeholder-shown=""
        data-scope="select"
        data-state="closed"
        dir="ltr"
        id="select::r0::trigger"
        role="combobox"
        type="button"
      >
        Servicio a realizar
        <span
          class="chakra-select__valueText css-176t9sc"
          data-part="value-text"
          data-scope="select"
          dir="ltr"
        />
      </button>
      <div
        class="chakra-select__indicatorGroup css-1xvjckd"
      >
        <div
          aria-hidden="true"
          class="chakra-select__indicator css-1eejzq5"
          data-part="indicator"
          data-scope="select"
          data-state="closed"
          dir="ltr"
        >
          <svg
            class="css-s3mb0o"
            viewBox="0 0 24 24"
          >
            <path
              d="m6 9 6 6 6-6"
            />
          </svg>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
